nsv13/code/modules/admin/tickets.dm

```diff
@@ -0,0 +1,194 @@
+			info["initiator"] = AH.initiator_key_name
+			info["antag_status"] = (AH.initiator.mob.mind && AH.initiator.mob.mind.special_role) ? AH.initiator.mob.mind?.special_role : "Non-antag"
+			info["ours"] = FALSE
+			info["open"] = TRUE
+			if(AH.tier == "mentor")
+				mentor["[AH.id]"] = info
+				continue
+			if(!check_rights(R_ADMIN)) //Double check that they're..y'know, allowed to see this one.
+				continue
+			else
+				unclaimed["[AH.id]"] = info
+		else //If we own the ticket, otherwise, split it up.
+			var/list/info = list()
+			info["id"] = AH.id
+			info["title"] = AH.name
+			info["initiator"] = AH.initiator_key_name
+			info["antag_status"] = (AH.initiator.mob.mind && AH.initiator.mob.mind.special_role) ? AH.initiator.mob.mind?.special_role : "Non-antag"
+			info["ours"] = TRUE
+			info["open"] = TRUE
+			ours["[AH.id]"] = info
+	if(show_resolved)
+		for(var/datum/admin_help/AH in GLOB.ahelp_tickets.closed_tickets)
+			var/list/info = list()
+			info["id"] = AH.id
+			info["title"] = AH.name
+			info["initiator"] = AH.initiator_key_name
+			info["antag_status"] = (AH.initiator.mob.mind && AH.initiator.mob.mind.special_role) ? AH.initiator.mob.mind?.special_role : "Non-antag"
+			info["ours"] = FALSE
+			info["open"] = FALSE
+			if(AH.tier != "mentor")
+				if(!check_rights(R_ADMIN)) //Double check that they're..y'know, allowed to see this one.
+					continue
+			resolved["[AH.id]"] = info
+	if(viewing)
+		var/list/info = list()
+		info["id"] = viewing.id
+		info["title"] = viewing.name
+		info["initiator"] = viewing.initiator_key_name
+		info["antag_status"] = (viewing.initiator.mob.mind && viewing.initiator.mob.mind.special_role) ? viewing.initiator.mob.mind?.special_role : "Non-antag"
+		data["currentInfo"] = info
+		data["open"] = (viewing.state == AHELP_ACTIVE) ? TRUE : FALSE
+		var/count = 0
+		var/flipflop = FALSE
+		for(var/X in viewing._interactions)
+			var/colour = flipflop ? "black" : "#696969"
+			if(findtext(X,"Resolved by"))
+				colour = "green"
+			else if(findtext(X, "Rejected by"))
+				colour = "red"
+			else if(findtext(X, "Closed by"))
+				colour = "red"
+			else if(findtext(X, "Marked as IC issue by"))
+				colour = "Average"
+			logs["[count]"] += list("line"=X, "colour"=colour)
+			flipflop = !flipflop
+			count ++
+	data["show_resolved"] = show_resolved
+	data["log"] = logs
+	data["screen"] = screen
+	data["unclaimed"] = unclaimed
+	data["resolved"] = resolved
+	data["mentor"] = mentor
+	data["ours"] = ours
+	return data
+
+/datum/admin_ticket_handler/ui_act(action, params)
+	if(..())
+		return
+	if(!holder.is_mentor())//Admins are by default counted as mentors
+		return
+	var/desiredID = text2num(params["id"])
+	var/datum/admin_help/AH = GLOB.ahelp_tickets.TicketByID(desiredID)
+	if(!AH)
+		AH = viewing
+	var/act = action
+	switch(action)
+		if("view")
+			viewing = AH
+			screen = "viewTicket"
+			return
+		if("back")
+			viewing = null
+			screen = "home"
+			return
+		if("re-class")
```

Comment:
Personally I suggest a IC button seperate from here because those are probably more often for admins.